graphistry/feature_utils.py

```diff
@@ -114,6 +146,33 @@ def assert_imported():
         raise import_min_exn
 
 
+def assert_cuml_cucat():
+    has_cuml_dependancy_, import_cuml_exn, cudf = lazy_import_has_dependancy_cu_cat()
+    if not has_cuml_dependancy_:
+        logger.error(  # noqa
+                     "cuml not found, trying running"  # noqa
+                     "`pip install rapids`"  # noqa
+        )
+        raise import_cuml_exn
+
+
+def make_safe_gpu_dataframes(X, y, engine):
+    has_cudf_dependancy_, _, cudf = lazy_import_has_dependancy_cu_cat()
```

Comment:
Add `assert cudf is not None` ?

Reply:
ok -- after the if statement seems best here again [like other assert you mentioned](https://github.com/graphistry/pygraphistry/pull/486#discussion_r1283943343)